### DISCUSSION TITLE
Is there a bot to detect new software updates? I notice other apps notify really fast for updates

### DISCUSSION THREAD

**Discussion by fernandog** [microsoft/winget-cli#2613] [General] [answered]
I always used another software to monitor software updates, then I found out Winget which is really great. I can't stop using the other one (SUMO) because it's really fast showing new updates (but have to manually install). Sometimes takes days or more to this apps update gets merged at Winget

1) So, is there a bot that detect new version or is only based on users Pull Requests? Or is there a bot that checks for some apps updates? How it works?
2) How come the other app detect updates so fast?! Is there a way to improve Winget on this matter?

**Comment by cgerke** [comment]
Just "Watch" the https://github.com/microsoft/winget-pkgs/ repo. You can filter your incoming emails with rules.

**Reply by Masamune3210** [reply]
If you do this, make sure you turn on pull requests and issues since I don't think the repo does discussions or releases

**Comment by Trenly** [comment, marked as answer]
> So, is there a bot that detect new version or is only based on users Pull Requests? 

Updates are only made available after there has been a pull request over at the winget-pkgs repository, it has passed the validation pipeline, and a moderator has approved the pull request

> Or is there a bot that checks for some apps updates? How it works?

There is wingetbot, which scans periodically for hash mismatches. When a mismatch is found, it attempts to update the package by submitting an automated pull request. There is also an external automation by @vedantmgoyal2009 which periodically scans packages that have ways to determine if a new version is available either through an API or web scraping. However, that only works for packages which have been added to the automation. Finally, publishers can integrate WingetCreate into their CI/CD pipelines and automatically open PRs when the release new versions - but adoption of this is low so far. All other updates rely on community contribution

> How come the other app detect updates so fast?! Is there a way to improve Winget on this matter?

Different package managers have different review and security processes. Winget is still relatively new as a package manager and is continuing to grow along with the community. The validation pipeline and malware scans are one of many tools that have been put in place to ensure security for the users, but which comes at the cost of slower updates. There are features requested for things like Verified Publishers, which would allow software providers to have a business account where manifests don’t need the moderator approval before merging, which can save some time, but will still be limited by the security scans.

The best way to improve app updates being available is to contribute the updates to the winget-pkgs repository. Its very easy to use a tool like YamlCreate or WingetCreate to submit an update if the package already exists in winget.

Ultimately, the goal is for publishers to submit updates as soon as they become available, which will only happen if there is enough push from the consumers of their application. If the application is open sourced, file an issue on their bug tracker as a feature request to integrate Winget as part of their CI/CD process; if its closed source, fill out their contact form or send an email. They might say no, in which case nothing changes, but they could also say yes. You never know unless you try
